File: CPP/kinematics.cpp
#include <iostream>

int main() {
    double initialPosition, initialVelocity, acceleration, time;

    // Input initial position, initial velocity, acceleration, and time
    std::cout << "Enter initial position: ";
    std::cin >> initialPosition;

    std::cout << "Enter initial velocity: ";
    std::cin >> initialVelocity;

    std::cout << "Enter acceleration: ";
    std::cin >> acceleration;

    std::cout << "Enter time: ";
    std::cin >> time;

    // Calculate final position using kinematic equation: s = ut + 0.5 * a * t^2
    double finalPosition = initialPosition + initialVelocity * time + 0.5 * acceleration * time * time;

    // Output the final position
    std::cout << "Final position: " << finalPosition << std::endl;

    return 0;
}

File: Python/kinematics.py
def main():
    initial_position = float(input("Enter initial position: "))
    initial_velocity = float(input("Enter initial velocity: "))
    acceleration = float(input("Enter acceleration: "))
    time = float(input("Enter time: "))
    
    final_position = calculate_final_position(initial_position, initial_velocity, acceleration, time)
    
    print("Final position:", final_position)

def calculate_final_position(initial_position, initial_velocity, acceleration, time):
    final_position = initial_position + initial_velocity * time + 0.5 * acceleration * time ** 2
    return final_position

if __name__ == "__main__":
    main()
